internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -272,95 +270,37 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	clusterName := d.Get("name").(string)
 	useReplicationSpecPerShard := false
 	var replicationSpecs []map[string]any
-	var clusterID string
 
 	if v, ok := d.GetOk("use_replication_spec_per_shard"); ok {
 		useReplicationSpecPerShard = v.(bool)
 	}
 
-	if !useReplicationSpecPerShard {
-		clusterDescOld, resp, err := connV220240530.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-		if err != nil {
-			if resp != nil {
-				if resp.StatusCode == http.StatusNotFound {
-					return nil
-				}
-				if admin20240530.IsErrorCode(err, "ASYMMETRIC_SHARD_UNSUPPORTED") {
-					return diag.FromErr(fmt.Errorf("please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details. %s", err))
-				}
```

Comment:
In previous implementation if cluster is asymmetric and `use_replication_spec_per_shard` is not defined, an error was returned providing guidance to enables this attribute.
I believe we are now not handling this validation, however it could be simple to incorporate.